### ISSUE TITLE
Broken as of version 4.3.0

### ISSUE THREAD

**Issue by olafur164** [epegzz/sass-vars-loader#29]
doing a map-get from a map inside js file is broken as most likely 4.3.0
currently i'm backing to 4.0.0 to get rid of this issue.

```
module.exports = {
  lightTheme: {
    background: 'white',
    color: "#000",
  },
}

```
```
@function color($map, $key, $alpha: 1) {
  $color: map-get($map, $key);
  @if $alpha == 1  {
    @return rgba($color, $alpha);
  }
  @else {
    @return $color;
  }
}

body {
  background-color: color($lightTheme, color);
}
```

`background-color: color($lightTheme, color);` returns background-color: "#000"; and therefore doesn't work..

i could do `unqote(map-get($map, $key)` but that doesn't work if i want to do `rgba`

**Comment by epegzz** [comment]
Thanks for reporting @olafur164, will fix it ASAP! :)

**Comment by olafur164** [comment]
@epegzz I was also wondering if there is any solution for getting the highlight in sass files.

so if i have a color variable `primary` imported from a javascript file i get in the editor in sass file that `$primary` exists

If not then me and a coworker of mine are thinking on working on an extension in VS Code for that.

**Comment by epegzz** [comment]
@olafur164 Syntax highlighting for variables in sass should be supported by the used sass plugin or even natively in VS Code. But it complains because it cannot find the definition of the variable, is that what you mean?

In any way, I'm myself an IntelliJ user, so my motivation to work on an VS Code plugin is a bit limited at this time :D

**Comment by olafur164** [comment]
What I was trying to say is that i don't get for example $primary highlight in sass file.

So i use your loader for importing variables from a javascript file, variables like colors, media breakpoints and etc so they are both available in the sass and javascript.
I then have https://github.com/shakacode/sass-resources-loader for loading other resources like functions and mixins.

then in the styles.scss file I do color: $primary and that works but i don't get the highlights that like if the variable was defined in a sass resource file instead of a javascript file.

doesn't give me any complains and works fine but I always have to check inside the colors.js file to check what the variable name is if i don't remember it.

Though maybe there is a plugin in VSCode already that solves this for me, just thought to ask :) But you're saying you get this in IntelliJ natively without a plugin?

**Comment by epegzz** [comment]
Nopes, I'm not getting this natively in IntelliJ either. This is a bit tricky and would definitely require a plugin :)

Hey BTW, you might not need the `sass-resource-loader`, you could simply add the files containing your mixins like that:

```
      { loader: "@epegzz/sass-vars-loader", options: {
          syntax: 'scss',
          files: [
            path.resolve(__dirname, 'config/my-mixins.sass')
          ]
        }
```

This does inline the mixins instead of adding an `@import`, but probably works as well :)

**Comment by epegzz** [comment]
@olafur164 fixed in `v4.3.1` :)

Now we only add quotes if the string starts with a white space or with a leading zero in order to still solve #27
